src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/Saml2SecurityTokenHandler.cs

```diff
@@ -0,0 +1,200 @@
+using Microsoft.IdentityModel.Tokens.Saml2;
+using MSIdentityTokens = Microsoft.IdentityModel.Tokens;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Creates SAML2 assertion-based security tokens
+    /// </summary>
+    public class Saml2SecurityTokenHandler : SecurityTokenHandler
+    {
+        /// <summary>
+        /// The key identifier value type for SAML 2.0 assertion IDs, as defined
+        /// by the OASIS Web Services Security SAML Token Profile 1.1. 
+        /// </summary>
+        public const string TokenProfile11ValueType = "http://docs.oasis-open.org/wss/oasis-wss-saml-token-profile-1.1#SAMLID";
+        private static readonly string[] s_tokenTypeIdentifiers = new string[] { SecurityTokenTypes.Saml2TokenProfile11, SecurityTokenTypes.OasisWssSaml2TokenProfile11 };
+        private SecurityTokenSerializer _keyInfoSerializer;
+        private readonly MSIdentityTokens.Saml2.Saml2SecurityTokenHandler _internalSaml2SecurityTokenHandler;
+        private readonly object _syncObject = new object();
+
+        /// <summary>
+        /// Creates an instance of <see cref="Saml2SecurityTokenHandler"/>
+        /// </summary>
+        public Saml2SecurityTokenHandler()
+           // : this(new Saml2SecurityTokenRequirement())
+        {
+            _internalSaml2SecurityTokenHandler = new MSIdentityTokens.Saml2.Saml2SecurityTokenHandler();
+        }
+
+        #region TokenValidation
+        /// <summary>
+        /// Returns value indicates if this handler can validate tokens of type
+        /// Saml2SecurityToken.
+        /// </summary>
+        public override bool CanValidateToken
+        {
+            get { return _internalSaml2SecurityTokenHandler.CanValidateToken; }
+        }
+
+        /// <summary>
+        /// Validates a <see cref="Saml2SecurityToken"/>.
+        /// </summary>
+        /// <param name="token">The <see cref="Saml2SecurityToken"/> to validate.</param>
+        /// <returns>The <see cref="ReadOnlyCollection{T}"/> of <see cref="ClaimsIdentity"/> representing the identities contained in the token.</returns>
+        /// <exception cref="ArgumentNullException">The parameter 'token' is null.</exception>
+        /// <exception cref="ArgumentException">The token is not assignable from <see cref="Saml2SecurityToken"/>.</exception>
+        /// <exception cref="InvalidOperationException">Configuration <see cref="SecurityTokenHandlerConfiguration"/>is null.</exception>
+        /// <exception cref="ArgumentException">Saml2SecurityToken.Assertion is null.</exception>
+        /// <exception cref="SecurityTokenValidationException">Thrown if Saml2SecurityToken.Assertion.SigningToken is null.</exception>
+        /// <exception cref="SecurityTokenValidationException">Thrown if the certificate associated with the token issuer does not pass validation.</exception>
+        public override ReadOnlyCollection<ClaimsIdentity> ValidateToken(SecurityToken token)
+        {
```

Comment:
This method is supposed to check Configuration.SaveBootstrapContext and if it's true, add it to the ClaimsIdentity. See [here](https://referencesource.microsoft.com/#System.IdentityModel/System/IdentityModel/Tokens/Saml2SecurityTokenHandler.cs,387) for the code.